### DISCUSSION TITLE
How to consume a GraphQL subscription from Ruby?

### DISCUSSION THREAD

**Discussion by Qqwy** [rmosolgo/graphql-ruby#4176] [Q&A]
Hi there!

We're trying to use GraphQL to send some data back-and-forth between our (Ruby on Rails, and some Elixir) servers. 
Subscriptions seem to be the natural tool for this task, and a good replacement for webhook-based approaches that we used in the past.

And besides actually communicating between servers, being able to run both client and server in the same app also seems like a great approach to testing that the subscriptions we set up trigger correctly.

The [documentation](https://graphql-ruby.org/subscriptions/implementation.html) lists how to use three transports (ActionCable/websockets, Pusher and Ably) together with two JS libraries (Apollo and Relay).

What options exist from consuming a GraphQL subscription from within Ruby?

**Comment by rmosolgo** [comment]
I've never heard it done, but it looks like there _are_ Ruby clients for those transports: 

- https://github.com/NullVoxPopuli/action_cable_client
- https://github.com/ably/ably-ruby
- https://github.com/pusher-community/pusher-websocket-ruby

I'm not really sure what the next step would be for integrating those into your app, but hopefully they're helpful!

**Reply by Qqwy** [reply]
Thank you!

The big question is of course how these could be wired up to one of the GraphQL client libraries that currently exist. It seems all of the ones I was able to find (graphql-client, gqli, graplient) currently are hardwired to use JSON-over-HTTP as transport.

<sub>(There also is graphql-ruby-client but it seems to have been deprecated a few years back)</sub>

I guess most people only use GraphQL between their back-end and front-end apps, rather than between back-ends, at least in Ruby-land. :man_shrugging:

**Reply by rmosolgo** [reply]
Oh, yeah -- I don't think any of the existing Ruby graphql clients support subscriptions at all 🙈
